src/net_processing.cpp

```diff
@@ -4241,6 +4246,8 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             return;
         }
 
+        if (m_txreconciliation) m_txreconciliation->TryRemovingFromSet(pfrom.GetId(), wtxid);
```

Comment:
In 3ec03ffa35a37625ff66fcf135f851c1220cde99: Shouldn't it be into `if (AlreadyHaveTx(GenTxid::Wtxid(wtxid)))`?

Reply:
This code should certainly go earlier than the `return` in that `if`. I will move it.
This looks like a bug, although it only brings some moderate corner-case inefficiency. I should see whether I can test it in latter PRs.